### ISSUE TITLE
window jumps to other screen after setting SDL_WINDOW_FULLSCREEN_DESKTOP

### ISSUE THREAD

**Issue by SDLBugzilla** [libsdl-org/SDL#1388]
# This bug report was migrated from our old Bugzilla tracker.

**Reported in version:** HG 2.1
**Reported for operating system, platform:** Linux, x86_64

# Comments on the original bug report:

On 2014-03-07 01:35:17 +0000, Ahab wrote:

> I have 2 screens and create a window on the one with ID 0, after setting SDL_WINDOW_FULLSCREEN_DESKTOP for this window, the window jumps to the screen with ID 1. It is not happening for the second screen.
> 
> That is the code iam using https://github.com/Jensus/ICWRenderingClient/blob/minimize/main.cpp
> 
> And this is the output of xrandr, to get an idea about the used displays:
> 
> Screen 0: minimum 320 x 200, current 3286 x 1080, maximum 32767 x 32767
> LVDS1 connected 1366x768+0+0 (normal left inverted right x axis y axis) 344mm x 194mm
>    1366x768       60.0*+
>    1024x768       60.0  
>    800x600        60.3     56.2  
>    640x480        59.9  
> VGA1 connected 1920x1080+1366+0 (normal left inverted right x axis y axis) 525mm x 297mm
>    1920x1080      60.0*+
>    1600x1200      60.0  
>    1680x1050      60.0  
>    1400x1050      60.0  
>    1600x900       60.0  
>    1280x1024      75.0     60.0  
>    1440x900       59.9  
>    1280x960       60.0  
>    1280x800       59.8  
>    1152x864       75.0  
>    1024x768       75.1     70.1     60.0  
>    832x624        74.6  
>    800x600        72.2     75.0     60.3     56.2  
>    640x480        75.0     72.8     66.7     60.0  
>    720x400        70.1  
> HDMI1 disconnected (normal left inverted right x axis y axis)
> DP1 disconnected (normal left inverted right x axis y axis)
> VIRTUAL1 disconnected (normal left inverted right x axis y axis)

On 2014-03-07 14:36:54 +0000, Ahab wrote:

> Iam not able to reproduce it with Windows 8.1 and it only happens on Linux when the screen sizes are not the same. The same code works fine with the following display setup.
> 
> xrandr
> Screen 0: minimum 320 x 200, current 3840 x 1080, maximum 8192 x 8192
> VGA-0 disconnected (normal left inverted right x axis y axis)
> HDMI-0 connected primary 1920x1080+0+0 (normal left inverted right x axis y axis) 521mm x 293mm
>    1920x1080      60.0*+   50.0     59.9     30.0     25.0     24.0     30.0     24.0  
>    1920x1080i     60.1     50.0     60.0  
>    1680x1050      59.9  
>    1280x1024      75.0     60.0  
>    1280x720       60.0     50.0     59.9  
>    1440x576i      50.1  
>    1024x768       75.1     60.0  
>    1440x480i      60.1     60.1  
>    800x600        75.0     60.3     56.2  
>    720x576        50.0  
>    720x480        60.0     59.9  
>    640x480        75.0     72.8     66.7     60.0     59.9  
>    720x400        70.1  
> DVI-0 connected 1920x1080+1920+0 (normal left inverted right x axis y axis) 525mm x 297mm
>    1920x1080      60.0*+
>    1600x1200      60.0  
>    1680x1050      59.9  
>    1400x1050      59.9  
>    1600x900       60.0  
>    1280x1024      75.0     60.0  
>    1440x900       59.9  
>    1280x960       60.0  
>    1280x800       59.9  
>    1152x864       75.0  
>    1024x768       75.1     70.1     60.0  
>    832x624        74.6  
>    800x600        72.2     75.0     60.3     56.2  
>    640x480        75.0     72.8     66.7     60.0  
>    720x400        70.1

**Comment by icculus** [comment]
To be verified (but not necessarily fixed) for 2.0.18

**Comment by icculus** [comment]
The original test code is gone, but this works for me here:

```c
#include <stdio.h>
#include "SDL.h"

int main(void) {
    SDL_Init(SDL_INIT_VIDEO);
    SDL_Window *window = SDL_CreateWindow(
        "Press F: Fullscreen (Red), Press W: Windowed (Green)",
        SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
        512, 512, SDL_WINDOW_RESIZABLE
    );

    SDL_Renderer *renderer = SDL_CreateRenderer(window, -1, 0);

    for (;;) {
        SDL_Event event;
        while (SDL_PollEvent(&event)) {
            switch (event.type) {
                case SDL_KEYDOWN: {
                    if (event.key.keysym.sym == SDLK_f) {
                        SDL_SetWindowFullscreen(window, SDL_WINDOW_FULLSCREEN_DESKTOP);
                    } else if (event.key.keysym.sym == SDLK_w) {
                        SDL_SetWindowFullscreen(window, 0);
                    }
                    break;
                }
                case SDL_QUIT: {
                    return 0;
                }
            }
        }

        if ((SDL_GetWindowFlags(window) & SDL_WINDOW_FULLSCREEN) != 0) {
            SDL_SetRenderDrawColor(renderer, 255, 0, 0, 255);
        } else {
            SDL_SetRenderDrawColor(renderer, 0, 255, 0, 255);
        }

        SDL_RenderClear(renderer);
        SDL_RenderPresent(renderer);
        SDL_Delay(1);
    }
}
```

Plugged my 1080p television into my laptop via HDMI, set the laptop's display to 1440x900 so it wasn't at the same resolution, ran that program, hit 'f' to fullscreen, 'w' to window...it stays on the display where the window is when the key is pressed. If I drag it to the other display, it still does the right thing when I go fullscreen.

So I guess we fixed this at some point.
